sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -202,7 +202,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
           DataSourceOptions.PATHS_KEY -> objectMapper.writeValueAsString(paths.toArray)
         }
         Dataset.ofRows(sparkSession, DataSourceV2Relation.create(
-          ds, extraOptions.toMap ++ sessionOptions + pathsOption,
+          ds, sessionOptions ++ extraOptions.toMap + pathsOption,
```

Comment:
Also, cc @rdblue since this is introduced at https://github.com/apache/spark/commit/aadf9535b4a11b42fd9d72f636576d2da0766199#diff-f70bda59304588cc3abfa3a9840653f4R197 .

Reply:
@dongjoon-hyun The commit didn't change semantic actually. Before it was:
```
(extraOptions ++
        DataSourceV2Utils.extractSessionConfigs(
          ds = ds.asInstanceOf[DataSourceV2],
          conf = sparkSession.sessionState.conf))
```

Reply:
Oh. It has more history. Thanks, @MaxGekk . Could you trace down when it started? We need to mark the affected version correctly in order to know the backport candidates.

Reply:
Also, please add a test case for this. If this has a long history than SPARK-23203 (fixed at 2.4.0), we need to verify this during backporting.

Reply:
The changes were added in https://github.com/apache/spark/pull/19861

Reply:
Thank you. Then, it was 2.3.0.